File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = defaultdict(int)
        output = []

        for x in nums:
            count[x] += 1

        while k > 0:
            max_key = max(count,key=count.get)            
            output.append(max_key)
            count.pop(max_key)
            k -= 1
        
        return output
        

        

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_set<char> char_set(s.begin(), s.end());
        int length = 0;

        for (char c : char_set) {
            int count = 0;
            int l = 0;
            for (int r = 0; r < s.size(); r++) {
                if (s[r] == c) {
                    count++;
                }
                
                while (r - l + 1 - count > k) {
                    if (s[l] == c) {
                        count--;
                    }
                    l++;
                }
                length = max(length, r - l + 1);
            }
        }
        return length;
    }
};
